### PR TITLE
Adding a specific Angular web performance guidance page.

### PR DESCRIPTION
**Pull request by Nooshu**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSSE-4984

### Change description ###
Added Angular specific Web Performance guidance as HMCTS has pockets of Angular usage within the department. This PR has been raised to compliment #454. If teams follow both sets of guidance, this will improve the overall service performance of all HMCTS services and any service that uses the Angular architecture.

This guidance includes topics such as:
- The use of Web Workers
- Server-Side Rendering (SSR)
- Static Site Generation (SSG)
- Weighing up the pros and cons of SSR vs SSG
- Known Anti-patterns to avoid
- How to fine-tune templates

### Checklist
- [x] this PR doesn't introduce any breaking changes.
- [x] commit messages are meaningful and follow good commit message guidelines.

### CODE REVIEW

**Review by linusnorton — APPROVED** [review]

File: source/standards/standards/angular-web-performance.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+There are some important considerations that should be thought about before using SSR. SSR increases the server-side workload, as the server needs to render each page for each user. This can impact scalability and requires careful server-side optimisation. Caching the rendered pages should be a high priority to reduce the server-side workload. It’s also important to consider SSR can add significant complexity to Angular applications.
+
+
+#### Summary
+
+Using Server-side Rendering in an Angular application comes with some trade-offs. But it can significantly improve performance, SEO, and user experience. These improvements are especially important for [public-facing websites and applications where performance](https://technology.blog.gov.uk/2019/04/18/why-we-focus-on-frontend-performance/) and SEO are crucial.
+
+
+### Static Site Generation (SSG)
```
Comment (timja) [inline source/standards/standards/angular-web-performance.html.md.erb:86]: Do we have any angular applications where this is applicable?

The content looks great but I'm concerned about these if we don't:

1. cognitive overhead of longer / more docs 
2. maintenance of additional docs 
3. risk of rot

**Review by timja — COMMENTED** [review]
Thanks for this it looks really good, I don't have much angular experience so I've mostly looked at this from a general point of view.

Would you mind checking my comments please, thanks

File: source/standards/standards/angular-web-performance.html.md.erb (hunk 1 above)
Comment (Nooshu) [reply source/standards/standards/angular-web-performance.html.md.erb:86]: I'm certainly willing to add these points you raise as questions within this guidance if that helps? e.g. "Before adopting SSR or SSG, evaluate if the added maintenance, documentation, and developer overhead outweigh potential benefits. Consider if the functionality risks becoming redundant."

Comment (Nooshu) [reply source/standards/standards/angular-web-performance.html.md.erb:88]: I've now added this feedback as a separate line under the considerations header.

**Review by timja — APPROVED** [review]
Last comment wasn't quite what I meant but its ok.

I'd suggest you join the HMCTS GitHub org and that way you won't need me to keep approving your CI checks

see https://hmcts.github.io/cloud-native-platform/onboarding/team/github.html#access

----

LGTM once CI checks pass

**Comment by linusnorton** [comment]
@timja waiting on https://github.com/hmcts/devops-azure-ad/pull/4227 :D

**Comment by timja** [comment]
> @timja waiting on [hmcts/devops-azure-ad#4227](https://github.com/hmcts/devops-azure-ad/pull/4227) :D

ha, approved.

**Comment by timja** [comment]
Merged separately in https://github.com/hmcts/hmcts.github.io/pull/458

**Pull request closed without merging**